users/service.go

```diff
@@ -814,3 +815,39 @@ func changed(updated *string, old string) bool {
 
 	return *updated != old
 }
+
+func (svc service) SendEmail(ctx context.Context, to []string, toType grpcEmailsV1.ContactType, from string, fromType grpcEmailsV1.ContactType, subject, header, user, content, footer string) error {
+	// Convert recipients based on contact type
+	emails := make([]string, len(to))
+	for i, contact := range to {
+		switch toType {
+		case grpcEmailsV1.ContactType_CONTACT_TYPE_ID:
+			u, err := svc.users.RetrieveByID(ctx, contact)
+			if err != nil {
+				return errors.Wrap(svcerr.ErrViewEntity, err)
+			}
+			emails[i] = u.Email
+		case grpcEmailsV1.ContactType_CONTACT_TYPE_EMAIL:
+			emails[i] = contact
+		default:
+			return errors.Wrap(svcerr.ErrMalformedEntity, errors.New("invalid contact type for recipients"))
+		}
+	}
+
+	// Convert sender based on contact type
+	var senderName string
+	switch fromType {
+	case grpcEmailsV1.ContactType_CONTACT_TYPE_ID:
+		inviter, err := svc.users.RetrieveByID(ctx, from)
```

Comment:
There should only be one `SendEmail` method (the same for the emailer) that encapsulates fetching ID if sender type is ID rather than email and so on. Also, encapsulate params to a DTO that's similar to the `protobuf` email request.